### PR TITLE
Add vale as a devcontainer feature

### PR DESCRIPTION
**Pull request by 14jdelap**
This references the issue about Vale.

**Problem**

Vale is installed but not found by `postCreateCommand`.

```
 - Installing vale (3.4.1.0)

Installing the current project: templatise (2.2.5)
/bin/sh: 1: vale: not found
[50042 ms] postCreateCommand failed with exit code 127. Skipping any further user-provided commands.
```

**Diagnosis**

Vale in `pyproject.toml` downloads a file but doesn't install it in `PATH`.

```
vscode ➜ /workspaces/template.py (main) $ find / -name vale
/workspaces/template.py/.venv/lib/python3.10/site-packages/vale
/workspaces/template.py/.venv/bin/vale

vscode ➜ /workspaces/template.py (main) $ echo $PATH
/vscode/vscode-server/bin/linux-x64/89de5a8d4d6205e5b11647eb6a74844ca23d2573/bin/remote-cli:/usr/local/python/current/bin:/usr/local/py-utils/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/vscode/.local/bin
```

**Solution**

I added Vale as a feature to `devcontainer.json`. This solves the installation problem and is a better location as it's a feature of devcontainer (rather than a dependency of whatever project someone starts using this template).

**Alternative considered**

Another approach would've been adding `/workspaces/template.py/.venv/bin/vale` to `PATH` in `devcontainer.json`. I don't think this is nearly as clean as a solution as the one I went with.

**How to test**

Create a DevContainer using this pull request and confirm that Vale is installed and `postCreateCommand` runs without errors.

### CODE REVIEW

**Comment by 14jdelap** [comment]
Done — ready for review. I think you need to approve for the workflow to execute @alunduil.

**Review by alunduil — APPROVED** [review]

**Comment by alunduil** [comment]
If you need help getting the tests to pass, let me know.

**Comment by 14jdelap** [comment]
`poetry run pytest` now passes locally.

I think I need your approval to run the workflow again (it'd be ideal if I can trigger it myself if that isn't a problem)

**Comment by alunduil** [comment]
@14jdelap I have it set up so that first time contributors require approval for workflow runs, so once this lands the next won't require any special approvals.

File: templatise_test/actions_test_fixtures/convert_file_devcontainer_json.yaml
Comment (alunduil) [inline templatise_test/actions_test_fixtures/convert_file_devcontainer_json.yaml:1]: This would be better as a disconnected golden test like I have in template.hs, if you're interested in making that change let me know and I'll write up an issue for it.

**Pull request merged**
